src/BloomExe/web/CommandAvailabilityPublisher.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Fleck;
+
+namespace Bloom.web
+{
+	/// <summary>
+	/// Commands have a stats that needs to be communicated to the html client.
+	/// This class does that by wiring up to events on commands and sending
+	/// updated status on commands over a websocket.
+	/// </summary>
+	class CommandAvailabilityPublisher : IDisposable
+	{
+		private readonly DuplicatePageCommand _duplicatePageCommand;
+		private WebSocketServer _server;
+		private List<IWebSocketConnection> _allSockets;
+
+		public CommandAvailabilityPublisher(IEnumerable<ICommand> commands )
+		{
+			
+			foreach (var command in commands)
+			{
+				command.EnabledChanged += command_EnabledChanged;
+			}
+
+			FleckLog.Level = LogLevel.Warn;
+			_allSockets = new List<IWebSocketConnection>();
+			_server = new WebSocketServer("ws://127.0.0.1:8189");
```

Comment:
Might be clearer, if there isn't some disadvantage, to use localhost instead of 127.0.0.1.